src/main/kubeconfig-manager.ts

```diff
@@ -24,14 +24,24 @@ export class KubeconfigManager {
   protected async init() {
     try {
       await this.contextHandler.ensurePort();
-      await this.createProxyKubeconfig();
+      this.tempFile = await this.createProxyKubeconfig();
     } catch (err) {
       logger.error(`Failed to created temp config for auth-proxy`, { err });
     }
   }
 
-  getPath() {
+  async getPath() {
+    // create proxy kubeconfig if it is removed
+    if ((this.tempFile || this.tempFile === "") && !(await fs.pathExists(this.tempFile))) {
```

Comment:
Why so complex a check on `tempFile`? Basically you are checking if `tempFile` is `null`, `undefined`, `0`, or `false`.

Reply:
`tempFile` is either string (containing path to Kubeconfig) or null. It's null when it's not generated or it's unlinked. After it's unlinked `getPath()` should return undefined, at least based on tests. So those are limitations we have. What do you suggest?

Reply:
From looking at the code `unlink` never gets called. If it can be `undefined` then the field should be marked as optional.

Finally, I would just do a `if (this.tempFile === undefined) { return undefined; }`. Though given all the uses of `getPath()` they all seem to be subtly broken in that none of them (that I can see) check for it being `undefined`.

Reply:
Refactored. And yes, you are right that unlink is not called anymore. This must be dropped off by some refactoring.